internal/server/apimigrator.go

```diff
@@ -23,26 +23,28 @@ type apiMigration struct {
 	method          string
 	path            string
 	version         string
-	redirect        string
+	rename          string
 	requestRewrite  func(c *gin.Context)
 	responseRewrite func(c *gin.Context)
 }
 
-func addRedirect(a *API, method, path, newPath, version string) {
-	// update any existing migrations with the legacy path
-	for i, mig := range a.migrations {
-		if len(mig.redirect) == 0 && mig.path == path {
+// TODO: rename to addRename
+func addRedirect(a *API, method, oldPath, newPath, version string) {
+	for i, existing := range a.migrations {
+		// update request/response rewrite migrations to the new canonical path
+		if existing.rename == "" && existing.path == oldPath {
 			a.migrations[i].path = newPath
 		}
-		if len(mig.redirect) > 0 && mig.redirect == path {
-			a.migrations[i].redirect = newPath
+		// update redirect migrations to redirect to the new canonical path
+		if existing.rename != "" && existing.rename == oldPath {
+			a.migrations[i].rename = newPath
 		}
```

Comment:
A couple things changed here, so calling them out.

The main change is to compare strings using `== ""` and `!= ""` instead of length checks. That should be a little more obvious that these are strings.

The other changes are renaming of variables: `path` -> `oldPath`, `redirect` -> `rename`.